packages/polling-controller/src/StaticIntervalPollingController.ts

```diff
@@ -48,19 +48,21 @@ function StaticIntervalPollingControllerMixin<TBase extends Constructor>(
 
       const key = getKey(networkClientId, options);
       const existingInterval = this.#intervalIds[key];
-      this._stopPollingByPollingTokenSetId(key);
```

Comment:
I thought this was needed as a guard to prevent multiple loops occurring?

Suppose I start polling, then start polling again, won't I now have two timeouts pending?

Reply:
`_startPollingByNetworkClientId` will not be called again with the same key by the [AbstractPollingController](https://github.com/MetaMask/core/blob/main/packages/polling-controller/src/AbstractPollingController.ts#L61), so this was unnecessary

Reply:
But that still means we're reliant on the caller to ensure we're not called more than we like.

If it costs us nothing to include it, is it a worthwhile guard to ensure that the `StaticPollingIntervalController` itself can enforce that no two loops can occur at once?

Reply:
Added back [here]([ad82cd8](https://github.com/MetaMask/core/pull/4230/commits/ad82cd80ab0502004c56bdcbfb5c7847468f292a)). I do want to point out that `_stopPollingByPollingTokenSetId` only prevents new loops from starting and does not cancel any loops that are still inflight